src/core/algorithms/ind/ind_algorithm.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <list>
+#include <string>
+#include <unordered_map>
+
+#include "algorithms/algorithm.h"
+#include "ind.h"
+#include "tabular_data/input_tableset_type.h"
+
+namespace algos {
+
+class INDAlgorithm : public Algorithm {
+protected:
+    config::InputTableSet input_table_set_;
+
+public:
+    struct DatasetInfo {
+        std::string table_name;
+        std::vector<std::string> header;
+    };
+    using DatasetsOrder = std::vector<DatasetInfo>;
+    using IND = model::IND;
+    using INDList = std::list<IND>;
+
+public:
+    explicit INDAlgorithm(std::vector<std::string_view> phase_names);
+
+    virtual DatasetsOrder const& GetDatasetsOrder() const = 0;
+    virtual INDList IndList() const = 0;
```

Comment:
It's better to use `INDList const&` return type in order to avoid unnecessary copying (in the case when the algorithm uses the same type and does not construct a new list of dependencies with each call)